File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.cpp
class Solution {
    void fill(
        vector<int>& candidates,
        vector<vector<int>>& res,
        vector<int>& combination,
        int ind,
        int target
    ) {
        if(target==0) {
            res.push_back(combination);
            return;
        }

        int i=ind;
        while(i<candidates.size() && target>=candidates[i]) {
            combination.push_back(candidates[i]);
            fill(candidates, res, combination, i+1, target-candidates[i]);
            combination.pop_back();
            int j=i;
            while(i<candidates.size() && candidates[i]==candidates[j]) i++;
        }
    }
public:
    vector<vector<int>> combinationSum2(vector<int>& candidates, int target) {
        sort(candidates.begin(), candidates.end());

        vector<vector<int>> res;
        vector<int> combination;
        fill(candidates, res, combination, 0, target);
        return res;
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r=nums.length-1, st=0;

        if(nums[l]>nums[r]) {
            while(l<=r) {
                int m = l + (r-l)/2;
                System.out.println(m);
                if(nums[m]<nums[0]) {
                    st=m;
                    r=m-1;
                } else {
                    l=m+1;
                }
            }
        }
        System.out.println(st);
        int n = nums.length;
        l = st;
        r = st+n-1;
        while(l<=r) {
            int m = l + (r-l)/2;
            if(nums[m%n]==target) return m%n;
            else if(nums[m%n]<target) l=m+1;
            else r=m-1;
        }
        return -1;
    }
}
